src/Schema/TableDiff.php

```diff
@@ -238,10 +248,34 @@ public function getAddedColumns(): array
         return array_values($this->addedColumns);
     }
 
-    /** @return list<ColumnDiff> */
+    /**
+     * @internal Use {@see getChangedColumns()} instead.
+     *
+     * @return list<ColumnDiff>
+     */
     public function getModifiedColumns(): array
     {
-        return array_values($this->changedColumns);
+        Deprecation::triggerIfCalledFromOutside(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/6080',
+            '%s is deprecated, you should use `getModifiedColumns()` instead.',
+            __METHOD__,
+        );
+
+        return array_values(array_filter($this->getChangedColumns(), static function (ColumnDiff $diff) {
+            return count($diff->changedProperties) > 0;
+        }));
+    }
+
+    /** @return list<ColumnDiff> */
+    public function getChangedColumns(): array
+    {
+        $legacyChangedColumns = [];
+        foreach ($this->renamedColumns as $oldName => $column) {
+            $legacyChangedColumns[$oldName] = new ColumnDiff($oldName, $column, []);
+        }
```

Comment:
This is backward compatibility for laravel which is using an internal (because it is public)
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Schema/Grammars/RenameColumn.php#L65C21-L68